jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/RemoveDataCaseFileInstanceCommand.java

```diff
@@ -60,6 +64,17 @@ public Void execute(Context context) {
         // apply authorization
         authorizationManager.checkDataAuthorization(caseFile.getCaseId(), caseFile, variableNames);
         
+        
+        org.jbpm.process.instance.ProcessInstance pi = (org.jbpm.process.instance.ProcessInstance) ksession.getProcessInstance(processInstanceId);
```

Comment:
logic spreaded between VariableScopeInstance and here.

Reply:
VariableScopeInstance is not really used in this scenario. Please check getVariable, which relies on CaseFileInnstan

Reply:
Remove logic is very particular of CaseFileInstance (the events published are specific), so there is not really logic spread besided the call to variableScopeInstance.isReadOnly, which I believe is probably fine.

Reply:
After some failed tests, I think the original code is good there, reusing VariableScopeInstance.setVariable cause case duplication because of the way the listeners are handled there. 
I do not think it is wise to change it more deeply.